src/devices/neptune_r900.c

```diff
@@ -0,0 +1,258 @@
+/** @file
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+ */
+
+/** @fn int r900_decode(r_device *decoder, bitbuffer_t * bitbuffer)
+Neptune r900 flow meter decoder.
+
+The product site lists E-CODER R900 amd MACH10 R900. Not sure if this decodes both.
+
+Tested on E-CODER R900 capture files.
+
+The device uses PPM encoding,
+- 1 is encoded as 30 us pulse.
+- 0 is encoded as 30 us gap.
+
+A gap longer than 320 us is considered the end of the transmission.
+
+The device sends a transmission every xx seconds.
+
+A transmission starts with a preamble of 0xAA,0xAA,0xAA,0xAB,0x52,0xCC,0xD2
+But, it is "zero" based, so if you insert a zero bit to the beginning of the bitstream,
+the preamble is:
+- 0x55,0x55,0x55,0x55,0xA9,0x66,0x69,0x65
+
+It should be sufficient to find the start of the data after 0x55,0x55,0x55,0xA9,0x66,0x69,0x65.
+
+Once the payload is decoded, the message is as follows:
+(from https://github.com/bemasher/rtlamr/wiki/Protocol#r900-consumption-message)
+- ID - 32 bits
+- Unkn1 - 8 bits
+- NoUse - 6 bits
+- BackFlow - 6 bits    // found this to be 2 bits in my case ???
+- Consumption - 24 bits
+- Unkn3 - 2 bits
+- Leak - 4 bits
+- LeakNow - 2 bits
+
+Some addidtional information here: https://github.com/bemasher/rtlamr/issues/29
+
+After decoding the bitstream into 104 bits of payload, the layout appears to be:
+
+Data layout:
+
+    IIIIIIII IIIIIIII IIIIIIII IIIIIIII UUUUUUUU ???NNNBB CCCCCCCC CCCCCCCC CCCCCCCC UU?TTTLL EEEEEEEE EEEEEEEE EEEEEEEE
+
+- I: 32-bit little-endian id
+- U:  8-bit Unknown1
+- N:  6-bit NoUse (3 bits)
+- B:  2-bit backflow flag
+- C: 24-bit Consumption Data
+- U:  2-bit Unknown3
+- T:  4-bit days of leak mapping (3 bits)
+- L:  2-bit leak flag type
+- E: 24-bit extra data????
+*/
+
+#include "decoder.h"
+
+int const map16to6[16] = { -1, -1, -1, 0, -1, 1, 2, -1, -1, 5, 4, -1, 3, -1, -1, -1 };
+
+static void decode_5to8(bitbuffer_t *bytes, uint8_t *base6_dec) {
+    // is there a better way to convert groups of 5 bits to groups of 8 bits?
+    for (int i=0; i < 21; i++) {
+        uint8_t data = base6_dec[i];
+        if (data < 0x0F) {
+            bitbuffer_add_bit(bytes, 0);
```

Comment:
Is the else case really needed?

Reply:
Nope. Sorry, I was still thinking in "string mode" when I was building 5-bit strings out of 4-bit numbers.

I fixed this in my latest commit.

Thank you for reviewing my code.

-Jeff.